lib/srv/app/aws/handler.go

```diff
@@ -137,6 +133,7 @@ func (s *SigningServiceConfig) CheckAndSetDefaults() error {
 //	 5. Sign HTTP request.
 //	 6. Forward the signed HTTP request to the AWS API.
 func (s *SigningService) RoundTrip(req *http.Request) (*http.Response, error) {
+	defer req.Body.Close()
```

Comment:
This is a RoundTrip function, not ServeHTTP.

```
$ go doc http.RoundTripper                                                                                                                                                                                       (base) 
package http // import "net/http"

type RoundTripper interface {
	// RoundTrip executes a single HTTP transaction, returning
	// a Response for the provided Request.
	//
	// RoundTrip should not attempt to interpret the response. In
	// particular, RoundTrip must return err == nil if it obtained
	// a response, regardless of the response's HTTP status code.
	// A non-nil err should be reserved for failure to obtain a
	// response. Similarly, RoundTrip should not attempt to
	// handle higher-level protocol details such as redirects,
	// authentication, or cookies.
	//
	// RoundTrip should not modify the request, except for
	// consuming and closing the Request's Body. RoundTrip may
	// read fields of the request in a separate goroutine. Callers
	// should not mutate or reuse the request until the Response's
	// Body has been closed.
	//
	// RoundTrip must always close the body, including on errors,
	// but depending on the implementation may do so in a separate
	// goroutine even after RoundTrip returns. This means that
	// callers wanting to reuse the body for subsequent requests
	// must arrange to wait for the Close call before doing so.
	//
	// The Request's URL and Header fields must be initialized.
	RoundTrip(*Request) (*Response, error)
}
```
> RoundTrip must always close the body, including on errors

Having said that, I am going to refactor the various places we implement RoundTripper to move the logic into http.Handler interface instead, because we're not following the RounderTripper interface requirements in multiple ways. As it turns out, the (current) implementation of oxy Forwarder avoids the issue by copying the request into RoundTrip, but we should not be breaking the interface contract anyway. I'm going to eventually do that refactoring in a separate PR to limit the scope of changes in this PR.